codec/encode.go

```diff
@@ -152,10 +152,12 @@ func marshalMessage(b []byte, pm proto.Message, deterministic bool) ([]byte, err
 	}
 
 	// lowest common denominator
-	bytes, err := proto.Marshal(pm)
-	if err != nil {
+	var buf proto.Buffer
+	buf.SetDeterministic(true)
```

Comment:
Since you've hardcoded deterministic to `true`, I believe you meant to leave this code alone and instead put this strategy inside the `if deterministic {` section.

Reply:
Done.